### PR TITLE
Implement Settings Page - Add view page

### LINKED ISSUE #11: [FEATURE]: Settings Page: Add show component
- [x] Refactor data provider to be more generic
- [x] Add Settings Page
- [x] Implement Show Settings

### PR DESCRIPTION
**Pull request by piouson**
- [x] Refactor data provider to be more generic
- [x] Add Settings Page
- [x] Implement Settings View
- [ ] Implement Settings Update

### CODE REVIEW

File: src/services/auth-provider.js [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
+        localStorage.setItem("systemToken", data.systemToken);
       });
   },
   logout: () => {
-    localStorage.removeItem('systemToken');
+    localStorage.removeItem("systemToken");
     return Promise.resolve();
   },
   checkError: (error) => {
     if (error.status === 401 || error.status === 403) {
```
Comment (simply-alliv) [inline src/services/auth-provider.js:27]: This line causes a `TypeError: Cannot read property 'status' of undefined` whenever I, as a user, try and save an updated settings page. 

NOTE: This is url on the browser address bar when on the edit settings page: `http://localhost:3000/comment-microapi-admin-dashboard#/Settings/undefined`.

Comment Dashboard - Google Chrome 2020_07_22 00_43_28 (2)

Comment (piouson) [reply src/services/auth-provider.js:27]: even when I added an `id` field, rendering settings page still generate this error.. 
image

Comment (piouson) [reply src/services/auth-provider.js:27]: so I've removed edit option from UI and will revisit later..
possibly, we can add all pages to get a clearer idea of fixes and then come back for any fixes left..?
this way we make ui progress

Comment (piouson) [reply src/services/auth-provider.js:27]: my guess is that I need to use a custom menu for the settings page but will leave this for now

Comment (piouson) [reply src/services/auth-provider.js:27]: this error shows why every record must have an `id` field..

Comment (simply-alliv) [reply src/services/auth-provider.js:27]: We can revisit as I don't have a full understanding of the error either.

**Pull request merged**